src/main/java/com/google/jenkins/plugins/k8sengine/KubernetesEnginePublisher.java

```diff
@@ -289,9 +289,27 @@ public ListBoxModel doFillZoneItems(
       }
     }
 
-    public FormValidation doCheckZone(@QueryParameter String value) {
-      if (Strings.isNullOrEmpty(value)) {
+    public FormValidation doCheckZone(
+        @AncestorInPath Jenkins context,
+        @QueryParameter("zone") String zone,
+        @QueryParameter("projectId") String projectId,
+        @QueryParameter("credentialsId") String credentialsId) {
+      if (Strings.isNullOrEmpty(zone)) {
         return FormValidation.error(Messages.KubernetesEnginePublisher_ZoneRequired());
+      } else if (Strings.isNullOrEmpty(projectId) || Strings.isNullOrEmpty(credentialsId)) {
+        return FormValidation.error(
```

Comment:
If we want to support the use case that someone can enter a zone that is not yet associated with the project id, these new messages could be warnings instead of errors. Thoughts?

Reply:
There are other ways to verify as well that I did not pursue: 

- storing a list of all GCP zones in a text file or something
    - Easily goes stale
- regex matching (e.g. `^\w\w+-\w\w+\d-\w$`) 
    - Potentially has the same problem if a new zone breaks the pattern

Reply:
I think the current approach is fine. I would err on avoiding approaches which add maintenance work for us in the future.